### PR TITLE
renepay: add self-pay feature

### LINKED ISSUE #6606: Support self-payment also for renepay
### Issue and Steps to Reproduce

```
lightning-cli renepay lnbc200n1pjwdlqfs...
{
   "code": -32602,
   "message": "This payment is destined for ourselves. Self-payments are not supported"
}
```

```
lightning-cli pay lnbc200n1pjwdlqfs...
{
   "destination": "02888244029c5909593038ab19f269947c720de3423e491791b46c7c92f76279b6",
   "payment_hash": "79fa46b20a5b2be34e54e53528bc162ad9956cef137da06ed8a15c6500689a06",
   "created_at": 1692859652.730,
   "parts": 1,
   "amount_msat": 20000,
   "amount_sent_msat": 20000,
   "payment_preimage": "938884dac9b69003760401d95d756bb87dbc447e3f541bf2069dc6b42f976a81",
   "status": "complete"
}
```
### `getinfo` output

"version": "v23.08"

### PR DESCRIPTION
**Pull request by Lagrang3**
Self-pay feature for `renepay`, based on the implementation of self-pay for the default `pay` plugin.

Fixes #6606

### CODE REVIEW

File: plugins/renepay/pay.c [hunk 1]
```diff
@@ -649,14 +706,17 @@ payment_listsendpays_previous(
 
 	json_for_each_arr(i, t, arr)
 	{
-		u64 partid, groupid;
-		struct amount_msat this_msat, this_sent;
+		u64 partid = 0, groupid;
+		struct amount_msat this_msat = AMOUNT_MSAT(0), this_sent;
 		const char *status;
```
Comment (cdecker) [inline plugins/renepay/pay.c:711]: Are these dummy initializations really needed? It prevents `valgrind` from tracking the initialization state of memory.

Comment (cdecker) [reply plugins/renepay/pay.c:711]: I see the need to `partid` below, however the `this_msat` variable may remain uninitialized.

File: plugins/renepay/pay.c [hunk 2]
```diff
@@ -760,6 +835,7 @@ payment_listsendpays_previous(
 	return send_outreq(cmd->plugin, req);
 }
 
+
```
Comment (cdecker) [inline plugins/renepay/pay.c:838]: Spurious whitespace change.

**Review by cdecker — APPROVED** [review]
Thanks @Lagrang3, I added this PR to the v24.02 milestone. Happy to merge as is, but added two minor nits for future changes.

ACK 75f226095a423d134f2891095ef59e28bccfedbf

File: plugins/renepay/pay.c (hunk 2 above)
Comment (Lagrang3) [reply plugins/renepay/pay.c]: Fixed c71628

File: plugins/renepay/pay.c (hunk 1 above)
Comment (Lagrang3) [reply plugins/renepay/pay.c]: Fixed c71628

**Comment by rustyrussell** [comment]
Folded fix commit, fixed up check-includes, rebased on master....

Ack ccb44453bdf605410c18e977841d09c37804ec12

File: plugins/renepay/pay.c [hunk 3]
```diff
@@ -624,14 +680,17 @@ payment_listsendpays_previous(
...
 	{
-		u64 partid, groupid;
+		u64 partid = 0, groupid;
 		struct amount_msat this_msat, this_sent;
 		const char *status;
 
-		// TODO(eduardo): assuming amount_msat is always known.
+		// TODO: we assume amount_msat is always present, but according
+		// to the documentation this field is optional. How do I
+		// interpret if amount_msat is missing?
```
Comment (rustyrussell) [inline plugins/renepay/pay.c:689]: amount_msat is never missing on payments renepay makes.  Older payments, sure...

**Review by rustyrussell — APPROVED** [review]

**Pull request merged**